### ISSUE TITLE
Custom service endpoint type not showing in the New Service Endpoint pulldown list

### ISSUE THREAD

**Issue by jsalado** [Microsoft/vsts-tasks#6623]
## Environment
TFS 2017 and VisualStudio Service: javiersalado.visualstudio.com

## Issue Description

I'm developing an extension for TFS and VSTS with custom build tasks that need to use a custom Service endpoint.

I have included the service endpoint contribution in my vss-extension.json, repackaged the extension and updated it in my on-prem TFS and in VSTS. I expected to see my service endpoint type in the of service endpoints in the new service endpoint pulldown. It is not happening. 

Find attached my vss-extension.json file (changed the extension to txt).
[vss-extension.txt](https://github.com/Microsoft/vsts-tasks/files/1789128/vss-extension.txt)

What am I missing?

Thank you in advance for your help,
J.

**Comment by jsalado** [comment]
I looked in the documentation for a complete reference and explanation of how to define a new service endpoint contribution. I found bits and pieces and relied a lot on examples of publicly available extension to put together my definition.

I would be great a more descriptive documentation and that you publish in this repo the contribution definition (in the vss-extension.json) of the tasks published here. I think that is missing to use them as examples.

Thanks,
J.

**Comment by jsalado** [comment]
Hello,

any further info about this? I need some help to move on.

Thanks,
J.

**Comment by GitHubSriramB** [comment]
@jsalado - Apologies for the delay. 

I see that the target you have defined for the service endpoint contribution is incorrect:
"targets": [
                "ms.vss-service.endpont-types"
            ],

Can you try this instead?
"targets": [ "ms.vss-endpoint.endpoint-types" ],

Did you check if the following blog might help in authoring custom service endpoint?
https://blogs.msdn.microsoft.com/sriramb/2016/09/15/service-endpoints-customization/

We have samples for custom service endpoints in vsts-rm-extensions GitHub repo. For e.g.
https://github.com/Microsoft/vsts-rm-extensions/blob/c7ca2a01b32c269344ac434e6f50dcd103137333/Extensions/BitBucket/Src/vss-extension.json

Pl. let me know if you are still having issues in registering your custom service endpoint.

**Comment by jsalado** [comment]
Hi Sriram,

Thank you for your feedback. I fixed the typo you mention (sorry for that, it was an obvious one I missed) and read through the blog and the examples you've linked (I did a couple of adjustments in the basic authentication inputDescriptors as well).

Unfortunately I still can't see my extension service endpoint in the list... either in our internal TFS nor in VSTS
Here you have attached my modified manifest.
[vss-extension.txt](https://github.com/Microsoft/vsts-tasks/files/1808304/vss-extension.txt)

I must be doing something else wrong... I just can't figure it out. Thank you again for your help.
J.

**Comment by omeshp** [comment]
@jsalado I can still see the targets as "ms.vss-service.endpont-types" in the attached extension json instead of "ms.vss-endpoint.endpoint-types". Are you sure you updated the targets? After changing targets to "ms.vss-endpoint.endpoint-types" and uploading and installing the extension, I was able to see the new endpoint in the dropdown list.

![image](https://user-images.githubusercontent.com/11783897/37460005-efb55bc8-286f-11e8-8fa6-50d2ac85d7e7.png)

**Comment by bryanmacfarlane** [comment]
Note that issues in this repo are for tasks in /tasks.  This should be a developer community issue

**Comment by jsalado** [comment]
@omeshp thank you for your help I just fixed the typo endpont to endpoint I didn't see the it was all wrong. I changed vss-service to vss-endpoint in the target and it works as expected. I'm very sorry to have bothered you with this, it was very obvious but sometimes the forest doesn't let you see the trees.

@bryanmacfarlane I'll post these kind of questions in the developer community from now on.

Thank you all
